source/agora/cli/version/main.d

```diff
@@ -0,0 +1,104 @@
+ }
+ +/
+/*******************************************************************************
+
+    Determine the version number that will be used during building the Agora node.
+
+    Version number is determined in the following order:
+    1. The 'AGORA_VERSION' environment variable will be used, if it is set to
+       a non-empty string.
+    2. Otherwise, the output of the `git describe` command will be used.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.cli.ver.main;
+
+import std.file : write, readText, exists;
+import std.format;
+import std.path;
+import std.process;
+import std.stdio : writeln;
+import std.string;
+
+import core.stdc.stdlib : exit;
+
+private immutable VersionFileName = "VERSION";
+private immutable RootPath = __FILE_FULL_PATH__.dirName.dirName.dirName.dirName.dirName;
+private immutable VersionFilePath = RootPath.buildPath("build").buildPath(VersionFileName);
+private immutable EnvVersionName = "AGORA_VERSION";
+private immutable GitDescribe = ["git", "describe"];
+
+/// Utility function to run a command and throw on error if the command fails
+private string runCmd (const string[] cmd)
+{
+    writeln("Executing command: ", cmd);
+    auto proc_res = execute(cmd);
+    if (proc_res.status != 0)
+        throw new Exception(format("Command failed: %s", cmd));
+    return proc_res.output;
+}
+
+/// Determine the version number
+private string getVersion()
+{
+    string version_res;
+
+    // get the version from environment variables
+    auto version_from_env = environment.get(EnvVersionName);
+    if (version_from_env !is null)
+    {
+        version_from_env = version_from_env.strip();
+        if (!version_from_env.length)
+            throw new Exception(format("Environment variable %s is empty, please remove it or set to non-empty value",
+                                EnvVersionName));
+        version_res = version_from_env;
+    }
+    else // get the version from git, if the version is not specified as an environment variable
+        version_res = runCmd(GitDescribe).strip(); // will throw if we are not in a git directory
+
+    if (!version_res.length) // in the very unlikely case of being in a git directory with no labels
+        throw new Exception("Unable to determine agora version number");
+
+    // get the version from the version file
+    string version_from_file;
+    if (VersionFilePath.exists)
+        version_from_file = VersionFilePath.readText().strip();
+
+    // if the version from the file is the same as the version deduced, then exit
+    if (version_from_file == version_res)
+        exit(0);
+
+    write(VersionFilePath, version_res);
+
+    return version_res;
+
+}
+
+///
+public void main(string[] args)
+{
+    try
+    {
+        getVersion();
```

Comment:
Since the return value is never used, and this really just writes the version, perhaps rename it, change the return value, and use `return` instead of `exit` in the above function ?